File: leetcode/all/go/429.go
package main

type Node struct {
	Val      int
	Children []*Node
}

func levelOrder(root *Node) [][]int {
	ans := [][]int{}
	if root == nil {
		return ans
	}
	q := []*Node{root}
	for len(q) > 0 {
		_len := len(q)
		_ans := []int{}
		for _, node := range q[:] {
			_ans = append(_ans, node.Val)
			q = append(q, node.Children...)
		}
		q = q[_len:]
		ans = append(ans, _ans)
	}
	return ans
}

File: archived/leetcode/all/python3/429.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""


from collections import deque
from typing import List


class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children


class Solution:
    def levelOrder(self, root: 'Node') -> List[int]:
        ans = []

        if not root:
            return ans

        q = deque()
        q.append(root)

        while q:
            n = len(q)
            res = []
            for _ in range(n):
                node = q.popleft()
                res.append(node.val)
                q.extend(node.children)
            ans.append(res)

        return ans
